### PR TITLE
Refactor RPC to Fully Utilize Ethereum APIs

### LINKED ISSUE #4105: Utilize Public Validator API Instead of Internal Endpoints
This is a tracking issue for us to use the Public Validator API instead of internal methods such as `RequestBlock` or `RequestAttestation` to maintain a nice standard for other clients to eventually interop with our API as well. Opening this after seeing issues such as https://github.com/prysmaticlabs/prysm/issues/4097 raised by users, which show we aren't using all of our public API endpoints to their full extent.

### PR DESCRIPTION
**Pull request by rauljordan**
This resolves #4105 

---

# Description

**Write why you are making the changes in this pull request**

This PR modifies Prysm to fully utilize Ethereum APIs instead of using an internal, Prysm only RPC service for the validator client. Instead, we use public endpoints such as `GetDuties`, `ProposeBlock`, maintained fully in the Ethereum APIs repository [here](https://github.com/prysmaticlabs/ethereumapis).

**Write a summary of the changes you are making**

There are little to no logic changes in this PR - only refactoring and ensuring our validator client fully utilizes Ethereum APIs for its entire flow.

### CODE REVIEW

File: beacon-chain/cache/attestation_data.go [hunk 1]
```diff
@@ -11,7 +11,6 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
```
Comment (rauljordan) [inline beacon-chain/cache/attestation_data.go]: No need to review, just changing argument types

File: beacon-chain/rpc/beacon/validators.go [hunk 2]
```diff
@@ -187,18 +188,24 @@ func (bs *Server) ListValidators(
 		requestedEpoch = q.Epoch
 	}
 
-	vals := headState.Validators
+	validatorList := make([]*ethpb.Validators_ValidatorContainer, 0)
```
Comment (rauljordan) [inline beacon-chain/rpc/beacon/validators.go]: Implementing: https://github.com/prysmaticlabs/ethereumapis/pull/75 (including validator indices in listvalidators response)

File: beacon-chain/rpc/beacon/validators.go [hunk 3]
```diff
@@ -537,3 +544,53 @@ func (bs *Server) GetValidatorQueue(
 		ExitPublicKeys:       exitQueueKeys,
 	}, nil
 }
+
+// GetValidatorPerformance reports the validator's latest balance along with other important metrics on
```
Comment (rauljordan) [inline beacon-chain/rpc/beacon/validators.go]: Moves this method to the beacon RPC as it makes more sense to have it there

Comment (rauljordan) [reply beacon-chain/rpc/beacon/validators.go]: No new logic

File: beacon-chain/rpc/service.go [hunk 4]
```diff
@@ -172,41 +170,25 @@ func (s *Service) Start() {
...
+		BlockReceiver:          s.blockReceiver,
 		MockEth1Votes:          s.mockEth1Votes,
-		AttPool:                s.attestationsPool,
-		CanonicalStateChan:     s.canonicalStateChan,
-		DepositFetcher:         s.depositFetcher,
+		Eth1BlockFetcher:       s.powChainService,
 		PendingDepositsFetcher: s.pendingDepositFetcher,
-		SyncChecker:            s.syncService,
-	}
-	attesterServer := &attester.Server{
```
Comment (rauljordan) [inline beacon-chain/rpc/service.go]: No need anymore for a separate attester/proposer server! we can have everything in the validator server itself

File: beacon-chain/rpc/validator/assignments.go [hunk 5]
```diff
@@ -3,22 +3,22 @@ package validator
...
 	"github.com/prysmaticlabs/prysm/beacon-chain/core/state"
 	pbp2p "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
-	pb "github.com/prysmaticlabs/prysm/proto/beacon/rpc/v1"
 	"github.com/prysmaticlabs/prysm/shared/bytesutil"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
-// CommitteeAssignment returns the committee assignment response from a given validator public key.
+// GetDuties returns the committee assignment response from a given validator public key.
```
Comment (rauljordan) [inline beacon-chain/rpc/validator/assignments.go]: Rename to reflect the public endpoint in Ethereum APIs named GetDuties

File: beacon-chain/rpc/validator/assignments_test.go [hunk 6]
```diff
@@ -7,20 +7,20 @@ import (
 	"sync"
```
Comment (rauljordan) [inline beacon-chain/rpc/validator/assignments_test.go]: No new logic in this file, just ensuring arguments use eth/v1alpha protos

File: beacon-chain/rpc/validator/attester.go [hunk 7]
```diff
@@ -1,94 +1,44 @@
...
-		}
-	}()
-
-	return &pb.AttestResponse{Root: root[:]}, nil
-}
-
-// RequestAttestation requests that the beacon node produce an IndexedAttestation,
-// with a blank signature field, which the validator will then sign.
-func (as *Server) RequestAttestation(ctx context.Context, req *pb.AttestationRequest) (*ethpb.AttestationData, error) {
+// GetAttestationData requests that the beacon node produce an attestation data object,
```
Comment (rauljordan) [inline beacon-chain/rpc/validator/attester.go]: Review this file - here we use the named methods from Ethereum APIs pretty much copying over logic we had in our attester RPC server into two functions: `GetAttestationData` and `ProposeAttestation`. No real logic changes.

File: beacon-chain/rpc/validator/attester_test.go [hunk 8]
```diff
@@ -1,4 +1,4 @@
-package attester
+package validator
```
Comment (rauljordan) [inline beacon-chain/rpc/validator/attester_test.go]: No need to review, just tidying up and ensuring things match valid args

File: beacon-chain/rpc/validator/proposer.go [hunk 9]
```diff
@@ -1,93 +1,63 @@
...
-var log logrus.FieldLogger
-
-func init() {
-	log = logrus.WithField("prefix", "rpc/proposer")
-}
-
-// Server defines a server implementation of the gRPC Proposer service,
-// providing RPC endpoints for computing state transitions and state roots, proposing
-// beacon blocks to a beacon node, and more.
-type Server struct {
```
Comment (rauljordan) [inline beacon-chain/rpc/validator/proposer.go]: We remove the old proposer server we used to have to instead just implement two public functions in this file: `GetBlock` and `ProposeBlock`

File: beacon-chain/rpc/validator/server.go [hunk 10]
```diff
@@ -99,57 +111,7 @@ func (vs *Server) ValidatorIndex(ctx context.Context, req *pb.ValidatorIndexRequ
 		return nil, status.Errorf(codes.Internal, "Could not find validator index for public key %#x not found", req.PublicKey)
 	}
 
-	return &pb.ValidatorIndexResponse{Index: index}, nil
-}
-
-// ValidatorPerformance reports the validator's latest balance along with other important metrics on
-// rewards and penalties throughout its lifecycle in the beacon chain.
-func (vs *Server) ValidatorPerformance(
```
Comment (rauljordan) [inline beacon-chain/rpc/validator/server.go]: Moved this to the beacon chain server instead

File: beacon-chain/rpc/validator/server_test.go [hunk 11]
```diff
@@ -24,7 +24,6 @@ import (
 	internal "github.com/prysmaticlabs/prysm/beacon-chain/rpc/testing"
```
Comment (rauljordan) [inline beacon-chain/rpc/validator/server_test.go]: No need to review, just ensuring tests use ethereumapis and pass

File: beacon-chain/rpc/validator/status.go [hunk 12]
```diff
@@ -6,9 +6,9 @@ import (
 	"math/big"
 	"time"
 
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
```
Comment (rauljordan) [inline beacon-chain/rpc/validator/status.go]: No logic changes here either, just ensuring params match ethereumapis

File: beacon-chain/rpc/validator/status_test.go [hunk 13]
```diff
@@ -16,7 +16,6 @@ import (
 	dbutil "github.com/prysmaticlabs/prysm/beacon-chain/db/testing"
```
Comment (rauljordan) [inline beacon-chain/rpc/validator/status_test.go]: No logic changes here as well, no need to review

File: third_party/com_github_prysmaticlabs_ethereumapis-tags.patch [hunk 14]
```diff
@@ -52,7 +52,7 @@ index a52dbad..33de299 100644
 +)
```
Comment (rauljordan) [inline third_party/com_github_prysmaticlabs_ethereumapis-tags.patch]: No need to review this, just the patch applied to include protobuf tags

File: validator/client/fake_validator_test.go [hunk 15]
```diff
@@ -66,10 +66,10 @@ func (fv *fakeValidator) NextSlot() <-chan uint64 {
 	return fv.NextSlotRet
 }
 
-func (fv *fakeValidator) UpdateAssignments(_ context.Context, slot uint64) error {
-	fv.UpdateAssignmentsCalled = true
-	fv.UpdateAssignmentsArg1 = slot
-	return fv.UpdateAssignmentsRet
+func (fv *fakeValidator) UpdateDuties(_ context.Context, slot uint64) error {
```
Comment (rauljordan) [inline validator/client/fake_validator_test.go]: Rename to match ethereumapis naming convention

File: validator/client/validator.go [hunk 16]
```diff
@@ -25,10 +25,9 @@ import (
 type validator struct {
```
Comment (rauljordan) [inline validator/client/validator.go]: No need to review, no logic changes - only using ethereumapis now

File: validator/client/validator_aggregate_test.go [hunk 17]
```diff
@@ -6,16 +6,17 @@ import (
 	"time"
```
Comment (rauljordan) [inline validator/client/validator_aggregate_test.go]: No need to review

File: validator/client/validator_attest.go [hunk 18]
```diff
@@ -10,7 +10,6 @@ import (
 	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
```
Comment (rauljordan) [inline validator/client/validator_attest.go]: No need to review

File: validator/client/validator_attest_test.go [hunk 19]
```diff
@@ -12,17 +12,16 @@ import (
 	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
```
Comment (rauljordan) [inline validator/client/validator_attest_test.go]: No need to review either

File: validator/client/validator_propose.go [hunk 20]
```diff
@@ -82,9 +81,17 @@ func (v *validator) ProposeBlock(ctx context.Context, slot uint64, pubKey [48]by
 	}).Info("Submitted new block")
 }
 
+// ProposeExit --
+func (v *validator) ProposeExit(ctx context.Context, exit *ethpb.VoluntaryExit) error {
```
Comment (rauljordan) [inline validator/client/validator_propose.go]: Stub until jim's changes are included

File: validator/client/validator_propose_test.go [hunk 21]
```diff
@@ -7,30 +7,23 @@ import (
```
Comment (rauljordan) [inline validator/client/validator_propose_test.go]: No need to review

**Comment by terencechain** [comment]
@rauljordan tests failing

**Comment by rauljordan** [comment]
Only e2e fails now

**Pull request merged**